src/etc/lldb_providers.py

```diff
@@ -44,32 +45,38 @@
 PY3 = sys.version_info[0] == 3
 
 
+class LLDBOpaque:
+    """
+    A type used by LLDB internally. Values of this type should never be used except when passing as
+    an argument to an LLDB function.
+    """
+
+    pass
```

Comment:
Nit, you don't need `pass` since there is a docstring.

Also I think that technically a global `LLDBOpaque = object` would be more accurate since the real type will be an `object` but will it will never be a `class LLDBOpaque`, but that doesn't matter much (If you leave the class, I would just make it clear in the docstring that this is strictly for typing).